template/source/javascripts/modules/in-page-navigation.js

```diff
@@ -0,0 +1,93 @@
+(function($, Modules) {
+  'use strict';
+
+  Modules.InPageNavigation = function InPageNavigation() {
+    var $tocPane;
+    var $contentPane;
+    var $tocItems;
+
+    var animationFrameRequested = false;
+
+    this.start = function start($element) {
+      if (!Modernizr.history) {
+        return;
+      }
+
+      $tocPane = $element.find('.app-pane__toc');
+      $contentPane = $element.find('.app-pane__content');
+      $tocItems = $tocPane.find('a');
+
+      $contentPane.on('scroll', function () {
+        withAnimationFrame(handleScrollEvent);
+      });
+
+      // Popstate is triggered when using the back button to navigate 'within'
+      // the page, i.e. changing the anchor part of the URL.
+      $(window).on('popstate', function (event) {
+        restoreScrollPosition(event.originalEvent.state);
+      });
+
+      // Restore state when e.g. using the back button to return to this page
+      restoreScrollPosition(history.state);
+    };
+
+    function restoreScrollPosition(state) {
+      if (state && state.scrollTop) {
+        $contentPane.scrollTop(history.state.scrollTop);
+      }
+    }
+
+    function handleScrollEvent() {
+      var activeTocItem = tocItemForFirstElementInView();
```

Comment:
Slightly pedantic, but as this returns a jQuery object, can we call this `$activeTocItem` for consistency? Appears to be what we're doing elsewhere.